File: SunnyFlamingo/Models/QuestionsBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SunnyFlamingo.Models
{
    public class QuestionsBase<T>
    {
        public T Value { get; set; }
        public string Key { get; set; }
        public int Order { get; set; }
        public List<QuestionBase<T>> QuestionBaseList { get; set; }
    }
}

File: SunnyFlamingo/ClientApp/src/app/models/questions-base.ts
import { QuestionBase } from "./question-base";

export class QuestionsBase<T> {
  value?: string;
  key: string;
  order: number;
  questionBaseList: QuestionBase<T>[];
  constructor(options: {
      value?: string;
      key?: string;
      order?: number;
      questionBaseList?: QuestionBase<T>[];
    } = {}){
    this.value = options.value;
    this.key = options.key || '';
    this.order = options.order === undefined ? 1 : options.order;
    this.questionBaseList = options.questionBaseList || [];
  }
}